src/herbpy/herbrobot.py

```diff
@@ -242,35 +319,187 @@ def CloneBindings(self, parent):
+            else:
+                logger.warning('The head is currently disabled under ros_control.')
+
+        # logic to determine which controllers are needed
+        if (self.right_arm in traj_manipulators and
+                not self.right_arm.IsSimulated() and
+                self.left_arm in traj_manipulators and
+                not self.left_arm.IsSimulated()):
+            controllers_manip.append('bimanual_trajectory_controller')
+        else:
+            if self.right_arm in traj_manipulators:
+                if not self.right_arm.IsSimulated():
+                    controllers_manip.append('right_trajectory_controller')
+                else:
+                    active_controllers.append(self.right_arm.sim_controller)
+
+            if self.left_arm in traj_manipulators:
+                if not self.left_arm.IsSimulated():
+                    controllers_manip.append('left_trajectory_controller')
+                else:
+                    active_controllers.append(self.left_arm.sim_controller)
+
+        # load and activate controllers
+        if not self.full_controller_sim:
+            self.controller_manager.request(controllers_manip).switch()
+
+        # repeat logic and actually construct controller clients
+        # now that we've activated them on the robot
+        if 'bimanual_trajectory_controller' in controllers_manip:
+            joints = []
+            joints.extend(self.right_arm.GetJointNames())
+            joints.extend(self.left_arm.GetJointNames())
+            active_controllers.append(
+                RewdOrTrajectoryController(self, '',
+                                           'bimanual_trajectory_controller',
+                                           joints))
+        else:
+            if 'right_trajectory_controller' in controllers_manip:
+                active_controllers.append(
+                    RewdOrTrajectoryController(self, '',
+                                               'right_trajectory_controller',
+                                               self.right_arm.GetJointNames()))
+
+            if 'left_trajectory_controller' in controllers_manip:
+                active_controllers.append(
+                    RewdOrTrajectoryController(self, '',
+                                               'left_trajectory_controller',
+                                               self.left_arm.GetJointNames()))
+
+        if needs_base:
+            if (hasattr(self, 'base') and hasattr(self.base, 'controller') and
+                    self.base.controller is not None):
+                active_controllers.append(self.base.controller)
+            else:
+                logger.warning(
+                    'Trajectory includes the base, but no base controller is'
+                    ' available. Is self.base.controller set?')
+
+        for controller in active_controllers:
+            controller.SetPath(traj)
+
+        if defer is True:
```

Comment:
We removed Trollius support from PrPy in https://github.com/personalrobotics/prpy/pull/278. It is no longer necessary to handle the `defer` argument.